xarray/core/accessors.py

```diff
@@ -0,0 +1,114 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from .common import is_datetime_like
+from .extensions import register_dataarray_accessor
+from .pycompat import dask_array_type
+
+import numpy as np
+import pandas as pd
+
+
+def _get_date_field(values, name):
+    """Indirectly access pandas' libts.get_date_field by wrapping data
+    as a Series and calling through `.dt` attribute.
+
+    Parameters
+    ----------
+    values : np.ndarray or dask.array-like
+        Array-like container of datetime-like values
+    name : str
+        Name of datetime field to access
+
+    """
+    def _access_through_series(values, name):
+        values_as_series = pd.Series(values.ravel())
+        field_values = getattr(values_as_series.dt, name).values
+        return field_values.reshape(values.shape)
+
+    if isinstance(values, dask_array_type):
+        from dask.array import map_blocks
+        out_dtype = object if name == "weekday_name" else np.int64
+        return map_blocks(_access_through_series,
+                          values, name, dtype=out_dtype)
+    else:
+        return _access_through_series(values, name)
+
+
+@register_dataarray_accessor('dt')
```

Comment:
Another thought: since this isn't being defined outside of xarray, it might actually make sense to import this class from `datarray.py` and simply add it to the DataArray class directly:
```
class DataArray:
    ...
    dt = property(DatatimeAccessor)
```

That has the advantage of being more transparent about whether it comes from.